sourcetracker/_sourcetracker.py

```diff
@@ -558,15 +599,25 @@ def gibbs_sampler(sink, cp, restarts, draws_per_restart, burnin, delay):
                     unknown_sum -= 1
 
                 # Calculate the new joint probability vector based on the
-                # removal of the ith sequence. Scale this probability vector
-                # for use by np.random.choice.
+                # removal of the ith sequence. Reassign the sequence to a new
+                # source environment and update counts for each environment and
+                # the unknown source if necessary.
+                # This is the fastest way I've currently found to draw from
+                # `jp`. By stacking (cumsum) the probability of `jp`, we can
+                # draw x from uniform variable in [0, total sum), and then find
+                # which interval that value lies in with searchsorted. Visual
+                # representation below
+                #          e1    e2  e3 e4  e5     unk
+                # jp:    |     |    |  |  |    |          |
+                # x:                          x
+                # new_e_idx == 4 (zero indexed)
+                # This is in contrast to the more intuitive, but much slower
+                # call it replaced:
+                # np.random.choice(num_sources, jp/jp.sum())
                 jp = cp.calculate_cp_slice(t, unknown_vector[t], unknown_sum,
                                            envcounts)
-
-                # Reassign the sequence to a new source environment and
-                # update counts for each environment and the unknown source
-                # if necessary.
-                new_e_idx = np.random.choice(source_indices, p=jp / jp.sum())
+                cs = jp.cumsum()
```

Comment:
Why the change from `jp.sum()` to `jp.cumsum()`? (Just want to make sure that that was intentional.)